lib/python/Pyxsim/__init__.py

```diff
@@ -71,6 +72,44 @@ def _build(xe_path, build_config=None, env={}, do_clean=False, clean_only= False
     return (success, output)
 
 
+def run_on_simulator_(xe, tester=None, simthreads=[], **kwargs):
+    
+    do_xe_prebuild = kwargs.get('do_xe_prebuild', False)
+    capfd = kwargs.get('capfd', None)
+    kwargs.pop("capfd")
+
+    if do_xe_prebuild:
+        build_env = kwargs.get('build_env', {})
+        do_clean = kwargs.get('clean_before_build', False)
+        build_success, build_output = _build(xe, env = build_env, do_clean = do_clean)
+
+        if not build_success:
+            return False
+
+    for k in ["do_xe_prebuild", "build_env", "clean_before_build"]:
+        if k in kwargs:
+            kwargs.pop(k)
+
+    Pyxsim.run_with_pyxsim(xe, simthreads, tester, **kwargs)
```

Comment:
Since this is in the same module file you should just be able to do `run_with_pyxsim([...])`. Also removes the need for the  import on line 16.

Reply:
Ta. C&P error.